src/main.c

```diff
@@ -357,6 +365,13 @@ static void focal_create_main_window(GApplication* app, FocalApp* fm)
 	gtk_widget_show_all(fm->mainWindow);
 
 	app_header_set_event(FOCAL_APP_HEADER(fm->header), NULL);
+
+	fm->sync_timer_id = gdk_threads_add_timeout_seconds_full(G_PRIORITY_DEFAULT_IDLE,
+															 120 /* sync interval, seconds */,
+															 (GSourceFunc) sync_func,
+															 fm,
+															 NULL /* finalize */);
```

Comment:
No need to involve gdk_threads. Suggest [g_timeout_add_seconds](https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#g-timeout-add-seconds)

Reply:
Okay, no problem -- I wasn't sure which approach to take and am new to gtk/glib programming :).  I actually used g_timeout_add originally I think, will switch back.  I don't remember what motivated switching-- I think it was in response to reading documentation text like this:

> You should use gdk_threads_add_idle() and gdk_threads_add_timeout() instead of g_idle_add() and g_timeout_add() since libraries not under your control might be using the deprecated GDK locking mechanism. If you are sure that none of the code in your application and libraries use the deprecated gdk_threads_enter() or gdk_threads_leave() methods, then you can safely use g_idle_add() and g_timeout_add().
(from https://developer.gnome.org/gdk3/stable/gdk3-Threads.html)

I have not worked with these details before, and reading bits like the above and other comments led me to believe it was "safer" to use gdk_threads to ensure appropriate locks are held (perhaps facilitating issues due to racing with other activities?).  Anyway I happily defer to almost anyone's suggestions, having little experience with this myself :).

Looking into things a bit further turns up instances such as[1] which sure makes it sound like your suggested approach is the way to go -- and gdk lock I thought was useful and safe is instead slated for removal? Hah, whoops!

Will update in a few, I believe I have the other approach on hand.

[1] https://gitlab.gnome.org/GNOME/gtk/commit/c655759cefa56efdb0cb943bb5dada3ff75a963c

Reply:
Oh, yes. I think I wasn't sure how to cleanly ensure the callback didn't just leak and only fired with a valid FocalApp pointer.

https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#mainloop-memory-management is what gave me some doubts.

(Apologies, as you can perhaps tell I have not worked with any GTK/glib applications before; much of what I know came from documentation and looking at examples in the gtk tree -- including a calendar sync one actually! haha)

Anyway I'll see what I can do.  If nothing else, maybe the primary benefit of this PR will be to let you know someone's interested and provide a smidge of motivation :grin: .  Ahem, one moment...

Reply:
> https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#mainloop-memory-management is what gave me some doubts.

The "first, and preferred" option mentioned there is exactly what you do in this PR - great.

Generally, focal is an asynchronous *single-threaded* application, so there should be no need to worry about threads at all